lib/livebook/hubs/dockerfile.ex

```diff
@@ -157,6 +157,45 @@ defmodule Livebook.Hubs.Dockerfile do
     |> Enum.join("\n")
   end
 
+  @doc """
+  Builds Dockerfile definition for Livebook Agent app deployment.
+  """
+  @spec build_agent_dockerfile(config(), Hubs.Provider.t()) :: String.t()
+  def build_agent_dockerfile(config, hub) do
+    base_image = Enum.find(Livebook.Config.docker_images(), &(&1.tag == config.docker_tag))
+
+    image = """
+    FROM ghcr.io/livebook-dev/livebook:#{base_image.tag}
+    """
+
+    image_envs = format_envs(base_image.env)
+
+    hub_config = """
+    ARG AGENT_KEY
+    ARG AGENT_NAME
+
+    # Teams Hub configuration for Livebook Agent deployment
+    ENV LIVEBOOK_AGENT_NAME ${AGENT_NAME}
+    ENV LIVEBOOK_TEAMS_KEY "#{hub.teams_key}"
+    ENV LIVEBOOK_TEAMS_AUTH "online:#{hub.hub_name}:#{hub.org_id}:#{hub.org_key_id}:${AGENT_KEY}"
```

Comment:
It would be nice if they could use the same Docker image with different envs when spawning different agents. With the current approach they will need to pass `LIVEBOOK_TEAMS_AUTH` as a whole, but that should be fine.

Looking at the whole Dockerfile I wonder if it actually makes sense to define a Dockerfile. Instead we could tell them to run `ghcr.io/livebook-dev/livebook:...` and list the env vars they should set. @josevalim?

Reply:
@jonatanklosko I think we can evolve to what you describe later, but while iterating, generating the Dockerfile is the most consistent approach for now.

Also, some places like Fly, require a Dockerfile, no?

Reply:
> Also, some places like Fly, require a Dockerfile, no?

You can specify an image instead :)

Whatever you think is better, we can always revisit, I am just pointing out that as a user not having to build the Dockerfile is easier, and using runtime envs rather than build args is preferable if possible.

Reply:
The issue is that CLI env vars show up in logs and stuff, which is why we went with build args or something?

Reply:
It depends on the setup, but build args don't have advantages over envs in this aspect. For example with fly you put build args in the config file, and you could put envs there too. But envs you can also put as fly secrets :)

The reason we use build args for the offline Dockerfile is because we do need the teams key at build time to run `warmup_apps.sh`.

Reply:
Makes sense for me to use env instead